imports/wasi_snapshot_preview1/poll_test.go

```diff
@@ -539,14 +528,44 @@ func fdReadSubFd(fd byte) []byte {
 // subscription for an EventTypeFdRead on stdin
 var fdReadSub = fdReadSubFd(byte(sys.FdStdin))
 
-type poller struct {
+// ttyStat returns fs.ModeCharDevice
+type ttyStat struct{}
+
+// Stat implements the same method as documented on platform.File
+func (ttyStat) Stat() (platform.Stat_t, syscall.Errno) {
+	return platform.Stat_t{
+		Mode:  fs.ModeDevice | fs.ModeCharDevice,
```

Comment:
as far as I know, we basically say it is a tty depending on if it is a char device not really anything else.

Reply:
Probably a fine approximation for most use cases, if we wanted to be more specific we could check with https://github.com/mattn/go-isatty/blob/v0.0.18/isatty_tcgets.go#LL11C1-L12C1

Reply:
I'll put a note for now. thanks for the link to add!